app/services/map_address_lookup_results.rb

```diff
@@ -1,19 +1,71 @@
 class MapAddressLookupResults
-  LINE_ONE_PARTS = %w[SUB_BUILDING_NAME BUILDING_NUMBER BUILDING_NAME].freeze
-  LINE_TWO_PARTS = %w[DEPENDENT_THOROUGHFARE_NAME THOROUGHFARE_NAME].freeze
+  NUMBER = %w[SUB_BUILDING_NAME BUILDING_NUMBER].freeze
+  NUMBER_NAME = %w[SUB_BUILDING_NAME BUILDING_NAME BUILDING_NUMBER].freeze
+  FULL_NAME = %w[SUB_BUILDING_NAME BUILDING_NAME].freeze
+  ROAD = %w[DEPENDENT_THOROUGHFARE_NAME THOROUGHFARE_NAME].freeze
 
   def self.call(results)
     results.map do |result|
       result['DPA'] && map_to_address(result['DPA'])
     end
   end
 
+  def self.building_name(result)
+    result['BUILDING_NAME']
+  end
+
+  def self.organisation_name(result)
+    result['ORGANISATION_NAME']
+  end
+
+  def self.building_number(result)
+    result['BUILDING_NUMBER']
+  end
+
+  def self.thoroughfare_name(result)
+    result['THOROUGHFARE_NAME']
+  end
+
+  def self.dependent_locality(result)
+    result['DEPENDENT_LOCALITY']
+  end
+
+  def self.combine(result, *attr)
+    result.slice(*attr).values.compact.join(', ')
+  end
+
+  def self.combine_with_road(result, *attr, separator)
+    [combine(result, *attr), combine(result, *ROAD)].compact.join(separator).strip
+  end
+
+  def self.use_dependent_locality?(result)
+    (building_name(result).nil? && organisation_name(result).nil?) ||
+      ((building_name(result).nil? || organisation_name(result).nil?) && thoroughfare_name(result).nil?)
+  end
+
+  def self.get_line_one(result)
+    return combine_with_road(result, *NUMBER_NAME, ' ') if building_name(result).nil? && organisation_name(result).nil?
+    return combine(result, *FULL_NAME) if organisation_name(result).nil?
```

Comment:
Duplication of condition with the line above e.g. organisation_name(result).nil?

Reply:
It's not quite duplication, the `organisation_name(result).nil?` needs to be present on both lines. 

The first `return` line checks whether the address has a building name or organisation name. If neither of those exists then the first line of the address is set as the building number and street name, and the method returns.

The second line will only be executed if one or both of the building name or organisation name are present. It checks to see if the organisation name is nil - if this is the case then we know that the building name is present, so the first line of the address is set to be the full building name, and the method returns.

If neither of those guards have returned then we know the address must have an organisation name present, and we always want the organisation on the first line, so the first line of the address is set to be the organisation name.

Reply:
Makes sense thanks 👍